crates/env/src/engine/off_chain/impls.rs

```diff
@@ -232,6 +232,20 @@ impl EnvInstance {
             .set_balance::<T>(dst_value + value)?;
         Ok(())
     }
+
+    fn terminate_contract_impl<T>(&mut self, beneficiary: T::AccountId) -> Result<()>
+    where
+        T: Environment,
+    {
+        // Send the remaining balance to the beneficiary
+        let all: T::Balance = self.balance::<T>().expect("could not decode balance");
+        self.transfer_impl::<T>(beneficiary, all)?;
+
+        let contract_id = self.account_id::<T>().expect("could not decode account id");
+        self.accounts.remove_account::<T>(contract_id);
+
```

Comment:
What the on-chain impl would do here is set a tombstone with a code hash and remove storage. AFAIU both is not easily achievable with our current off-chain env, hence I left it out here for the moment.